diff --git a/src/core/mac/mac.cpp b/src/core/mac/mac.cpp
index 6997fd01b..1e9903a3c 100644
--- a/src/core/mac/mac.cpp
+++ b/src/core/mac/mac.cpp
@@ -54,6 +54,21 @@
 #include "thread/thread_netif.hpp"
 #include "thread/topology.hpp"

+#if CONFIG_OPENTHREAD_CSMABACKOFF_OPTIMIZATION
+bool get_pm_standby_flag(void)
+{
+    bool pm_flag;
+    #if CONFIG_SOC_SERIES_RISCV_TELINK_B9X_RETENTION
+    extern bool b9x_deep_sleep_retention;
+    pm_flag = b9x_deep_sleep_retention;
+    #elif CONFIG_SOC_SERIES_RISCV_TELINK_TLX_RETENTION
+    extern bool tlx_deep_sleep_retention;
+    pm_flag = tlx_deep_sleep_retention;
+    #endif
+    return pm_flag;
+}
+#endif
+
 namespace ot {
 namespace Mac {

@@ -601,6 +616,13 @@ void Mac::StartOperation(Operation aOperation)

     if (mOperation == kOperationIdle)
     {
+#if CONFIG_OPENTHREAD_CSMABACKOFF_OPTIMIZATION
+    bool ot_deep_sleep_flag = !get_pm_standby_flag();
+    if(ot_deep_sleep_flag){
+            ot_deep_sleep_flag = false;
+            PerformNextOperation();
+    }else
+#endif /* CONFIG_OPENTHREAD_CSMABACKOFF_OPTIMIZATION */
         mOperationTask.Post();
     }
 }
diff --git a/src/core/mac/sub_mac.cpp b/src/core/mac/sub_mac.cpp
index 7d7ac1750..cad4cc48c 100644
--- a/src/core/mac/sub_mac.cpp
+++ b/src/core/mac/sub_mac.cpp
@@ -358,8 +358,12 @@ Error SubMac::Send(void)
 #if OPENTHREAD_CONFIG_MAC_ADD_DELAY_ON_NO_ACK_ERROR_BEFORE_RETRY
     mRetxDelayBackOffExponent = kRetxDelayMinBackoffExponent;
 #endif
-
+
+#if CONFIG_OPENTHREAD_CSMABACKOFF_OPTIMIZATION
+    StartCsmaFirstNoBackoff();
+#else
     StartCsmaBackoff();
+#endif

 exit:
     return error;
@@ -454,6 +458,60 @@ exit:
     return;
 }

+#if CONFIG_OPENTHREAD_CSMABACKOFF_OPTIMIZATION
+void SubMac::StartCsmaFirstNoBackoff(void)
+{
+    uint8_t backoffExponent = kCsmaMinBe + mCsmaBackoffs;
+
+#if !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+    if (mTransmitFrame.mInfo.mTxInfo.mTxDelay != 0)
+    {
+        SetState(kStateCslTransmit);
+
+        if (ShouldHandleTransmitTargetTime())
+        {
+            if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) <
+                Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) + mTransmitFrame.mInfo.mTxInfo.mTxDelay -
+                    kCcaSampleInterval - kCslTransmitTimeAhead)
+            {
+                mTimer.StartAt(Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) - kCcaSampleInterval -
+                                   kCslTransmitTimeAhead,
+                               mTransmitFrame.mInfo.mTxInfo.mTxDelay);
+            }
+            else // Transmit without delay
+            {
+                BeginTransmit();
+            }
+        }
+        else
+        {
+            BeginTransmit();
+        }
+
+        ExitNow();
+    }
+#endif // !OPENTHREAD_MTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+
+    SetState(kStateCsmaBackoff);
+
+    VerifyOrExit(ShouldHandleCsmaBackOff(), BeginTransmit());
+
+    if (backoffExponent > kCsmaMaxBe)
+    {
+        backoffExponent = kCsmaMaxBe;
+    }
+
+    if(otThreadGetDeviceRole(otInstanceInitSingle()) == OT_DEVICE_ROLE_CHILD)
+        HandleTimer();
+    else
+    StartTimerForBackoff(backoffExponent);
+
+exit:
+    return;
+}
+#endif
+
+
 void SubMac::StartTimerForBackoff(uint8_t aBackoffExponent)
 {
     uint32_t backoff;
diff --git a/src/core/mac/sub_mac.hpp b/src/core/mac/sub_mac.hpp
index e56321058..f46579e39 100644
--- a/src/core/mac/sub_mac.hpp
+++ b/src/core/mac/sub_mac.hpp
@@ -617,6 +617,7 @@ private:
     void ProcessTransmitSecurity(void);
     void SignalFrameCounterUsed(uint32_t aFrameCounter, uint8_t aKeyId);
     void StartCsmaBackoff(void);
+    void StartCsmaFirstNoBackoff(void);
     void StartTimerForBackoff(uint8_t aBackoffExponent);
     void BeginTransmit(void);
     void SampleRssi(void);
